lms/djangoapps/teams/static/teams/js/views/edit_team.js

```diff
@@ -119,6 +119,8 @@
 
                     this.teamModel.save(data, { wait: true })
                         .done(function(result) {
+                            // So that the topics list can be refreshed.
+                            Backbone.trigger('teamModel:add', view.teamModel);
```

Comment:
I'm coming around to the notion of a high level event that we fire when a team is created. It seems that we can't easily update all the models that might have been affected by this update, but equally we don't want to perform a fetch on all those models at once. We should have every view listen for these global events, and they should clear out any stale models at that point. Then when such a view is re-shown to the user, it should perform a fetch at that point to update its data. For example, the user's team membership collection should be refreshed, but only when we show a view that needs it (i.e. the "My Teams" view). To my mind, the key thing is that the views are in charge of keeping themselves fresh.

I'm going to write this up as an email to summarize it for the team. Let me know what you think of this approach.